com/course/zenkouv/src/Lecture3/Task1/Knight.java

```diff
@@ -0,0 +1,124 @@
+package Lecture3.Task1;
+
+import Lecture3.Task1.comparator.CostComparator;
+import Lecture3.Task1.comparator.WeightComparator;
+import Lecture3.Task1.model.Armor;
+import Lecture3.Task1.model.Outfit;
+import Lecture3.Task1.model.Weapon;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class Knight {
+
+    private final String name;
+    private final List<Outfit> outfitList;
+    private int outfitsCost;
+
+    public Knight(String name) {
+        this.name = name;
+        this.outfitList = new ArrayList<>();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public List<Outfit> getOutfitList() {
+        return outfitList;
+    }
+
+    public int getOutfitsCost() {
+        return this.outfitsCost;
+    }
+
+    public void addOutfitToOutfitLis(Outfit outfit) {
+        if (!availabilityOutfit(outfit)) {
+            outfitList.add(outfit);
+        }
+    }
+
+    private boolean availabilityOutfit(Outfit outfit) {
+        if (isWeapon(outfit)) {
+            for (Outfit outfitFromList : outfitList) {
+                if (outfitFromList instanceof Weapon) {
+                    System.out.printf("I have a %s and me don't need any weapon!\n",
+                            ((Weapon) outfitFromList).getType().toString().toLowerCase());
+                    return true;
+                }
+            }
+        } else {
+            for (Outfit outfitFromList : outfitList) {
+                if (outfitFromList instanceof Armor) {
+                    if (((Armor) outfit).getType() == ((Armor) outfitFromList).getType()) {
+                        System.out.printf("I have a %s and me don't need any %s!\n",
+                                ((Armor) outfitFromList).getType().toString().toLowerCase(),
+                                ((Armor) outfitFromList).getType().toString().toLowerCase());
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    private boolean isWeapon(Outfit outfit) {
+        return outfit instanceof Weapon;
+    }
+
+    public void calculateOutfitsCost() {
+        if (haveAnyOutfits()) {
+            outfitsCost = outfitList.stream().mapToInt(Outfit::getCost).sum();
+        }
+    }
+
+    private boolean haveAnyOutfits() {
+        if (this.outfitList.isEmpty()) {
+            System.out.println("I haven't any outfit!");
+            return false;
+        }
+        return true;
+    }
+
+    public void compareOutfitsByCost() {
+        outfitList.sort(new CostComparator());
```

Comment:
create custom  instance of comparator it is good way if you need some custom comparing
in case if you compare using one int filed use static factory method 
 outfitList.sort(Comparator.comparingInt(Outfit::getCost));

Reply:
fix